pkg/recommender.py

```diff
@@ -145,6 +147,19 @@ def dispatch(self, speaker_id, evt):
         if not isinstance(evt, np.ndarray):
             evt = np.array(evt)
 
+        #safety in case of numpy array or float
+        speaker_id = int(speaker_id)
```

Comment:
i don't think we can really take care of it. For example, what if the input is an invalid string, like `abcdef`? Convert it to `int` will crash our program. If you want to have a guard to protect us, we can simply raise an explicit error to the caller if it is not a integer, so the caller will receive a clear error message and it is the caller's responsibility to prepare correct input anyway
```py
if type(speaker_id) is not int:
  raise Error('Speaker id must be integer, received: ' + speaker_id)
```

Reply:
fyi, @kaiyingshan , just ensure the `grpc` interface set the right type for `speaker_id`

Reply:
oh ok that is a much better solution, thanks. I will make the change

Reply:
I was thinking about this some more, if we raise an error then our system will also crash. Could I just log the speaker id is invalid then return from the dispatch function? This way if the speaker id is corrected from Ashley's side, then our system will not have to be restarted.

Reply:
i mean you can, but would be totally pointless, since the speaker id is wrong, we cannot really do anything.
for such an unacceptable bug, i feel we can just expose it as early as possible so the responsible party can be informed during the development.